### DISCUSSION TITLE
stellar population synthesis model parameters for galaxies

### DISCUSSION THREAD

**Discussion by ilsangyoon** [LSSTDESC/desc-data-portal#81] [Q&A] [answered]
Hello,

My name is Ilsang Yoon and I am LSST DP0 delegee. I am interested in how well the existing or currently developed analysis tools recover the physical properties of galaxies observed by LSST. One of the interesting properties is stellar population synthesis model parameters including star formation history, for each galaxy. The people from LSST DP0 confirmed that the DP0 products do not have such information and suggest that I contact the LSST DESC team with the question. Can we access the stellar population synthesis model parameters used to create galaxy images in DP0, for each individual galaxy?

Thank you,

**Comment by evevkovacs** [comment, marked as answer]
The short answer is no. The M* and SFR values of each galaxy were determined from an empirical model (UniverseMachine). @aphearin may have additional comments about recovering information from UniverseMachine galaxies.  Please see https://arxiv.org/pdf/1907.06530.pdf for a detailed description of how the extragalactic catalog was produced.

**Comment by aphearin** [comment]
Hi @ilsangyoon I think @evevkovacs is right that the current mock does not have the precise relationship between star formation history and SED that would be needed for the sort of physical analysis you describe. We are working on modeling extensions that will supply the required relationship in future work, but the existing mock the relationship is "too empirical" for such purposes.

**Reply by ilsangyoon** [reply]
Thanks, Eve and Andrew

After reading several papers and documents and papers, my understanding is that LSST DP0 product is from DC2 sky simulation following LSST observing strategy and the galaxies in the DC2 simulation are from CosmoDC2, where I can find all true input parameters including stellar population synthesis models by cross-matching DC2 galaxies to the one in CosmoDC2 using Truth match table in DC2 catalog. Do I understand it correctly?

**Reply by yymao** [reply]
@evevkovacs can confirm if the input parameters to the stellar population synthesis model are included in the cosmoDC2 data files. 

If those parameters are indeed included, then, yes, @ilsangyoon your understanding is correct.  In particular, in the Truth-match Table there are `cosmodc2_hp` and `cosmodc2_id` which you can use to find the corresponding entries in the cosmoDC2 data set.  The cosmoDC2 is also available via the data portal (https://lsstdesc-portal.nersc.gov/doc/cosmodc2)

**Reply by ilsangyoon** [reply]
Hi @yymao, and @evevkovacs  and @aphearin 

Very recently I started working on this and checked whether I can actually find the matched information from CosmoDC2 catalog. I selected few objects from the truth-match catalog in the Rubin science platform and used their `cosmodc2_hp` and `cosmodc2_id` to find out the object in the CosmoDC2 catalog (I only used a subset of the catalog with HEALPix id) and I see the following keys and values are available. So, I think I can get the truth input parameters for each galaxy. I will need to read the CosmoDC2 paper to understand how these parameters are defined and used to create mock images. Thanks for your help. 

<KeysViewHDF5 ['LSST_filters', 'SDSS_filters', 'SEDs', 'baseDC2', 'blackHoleAccretionRate', 'blackHoleEddingtonRatio', 'blackHoleMass', 'convergence', 'dec', 'dec_true', 'diskAbundancesStellarMetals', 'diskMassStellar', 'diskStarFormationRate', 'diskVelocity', 'emissionLines', 'galaxyID', 'hostHaloMass', 'hostHaloTag', 'infallIndex', 'isCentral', 'lightcone_replication', 'lightcone_rotation', 'magnification', 'matchUp', 'morphology', 'otherLuminosities', 'peculiarVelocity', 'ra', 'ra_true', 'redshift', 'redshiftHubble', 'shear1', 'shear2', 'spheroidAbundancesStellarMetals', 'spheroidMassStellar', 'spheroidStarFormationRate', 'spheroidVelocity', 'spinSpin', 'step', 'totalAbundancesStellarMetals', 'totalMassStellar', 'totalStarFormationRate', 'totalVelocity', 'uniqueHaloID', 'vx', 'vy', 'vz', 'x', 'y', 'z']>

**Reply by evevkovacs** [reply]
Yes, you can get these from cosmoDC2.  (You should note that this list is for native catalog quantities, not schema (derived) quantities.) Let us know if you have any questions.

**Reply by yymao** [reply]
To supplement Eve's comment, it may be easier to access cosmoDC2 using `GCRCatalogs` -- you can find the instructions here: https://lsstdesc-portal.nersc.gov/doc/cosmodc2

If you use GCRCatalogs, you can also find the schema here: https://github.com/LSSTDESC/gcr-catalogs/blob/master/GCRCatalogs/SCHEMA.md#extragalactic-catalogs

**Reply by ilsangyoon** [reply]
Thanks @evevkovacs and @yymao I used a single hdf5 file because the entire CosmoDC2 catalog is too big to store on my laptop. So I use h5py. Can I use GCRCatalogs to read hdf5 file, or is there a better way of handling individual hdf5 file?

**Reply by yymao** [reply]
You can use `GCRCatalogs` even if you only have downloaded a subset of cosmoDC2. The code will only load the ones you have downloaded.

**Comment by evevkovacs** [comment]
If you are happy using h5py, that is fine. If you look at the documentation above, you will see that GCRCatalogs provides a reader that gives certain quantities with standardized names in particular units, so you can also use GCRCatalogs as an alternative way to read cosmoDC2. You will have to download and install the GCRCatalogs package. The reader allows you to read in specific subsets of healpixels which can be convenient for testing code.

**Reply by ilsangyoon** [reply]
Hi @yymao and @evevkovacs 

I have installed GCRCatalog and followed the setup instruction. But even though I set the path as shown in the instruction (directory /Users/iyoon/Science/LSST that includes lsstdesc-public as a subdirectory and under lsstdesc-public there is dc2 directory where actually datafiles are stored. The lsstdesc-public and dc2 are created automatically when I download the data via Globus), simple command like `gc = GCRCatalogs.load_catalog('cosmoDC2_v1.1.4')` returns error message "Catalog directory /Users/iyoon/Science/LSST/xgal/cosmoDC2/cosmoDC2_v1.1.4_rs_scatter_query_tree_double does not exist"

Why does GCRCatalog have appended directory name: xgal/cosmoDC2, which I didn't have or specify?

**Reply by yymao** [reply]
@ilsangyoon This is due to the different directory structures for internal/public data sets. Try loading cosmoDC2 with:
```
gc = GCRCatalogs.load_catalog('desc_cosmoDC2') 
```

It's the same catalog but this one points to `lsstdesc-public/dc2/cosmoDC2_v1.1.4`

**Reply by ilsangyoon** [reply]
@yymao Thanks for quick response. Now everything seems fine. So, later if I add more HEALPix files (hdf5) to the current directory, it will automatically include everything (sub files) in the directory, correct?

**Reply by yymao** [reply]
Yes!

**Reply by yymao** [reply]
But you do need to re-run the `GCRCatalogs.load_catalog` line for it to notice the newly added files.

**Reply by ilsangyoon** [reply]
Thank you! @yymao
